cves/kernel/CVE-2019-19045.yml

```
@@ -387,38 +413,39 @@ lessons:
     If you think of another lesson we covered in class that applies here, feel
     free to give it a small name and add one in the same format as these.
   defense_in_depth:
-    applies:
-    note:
+    applies: true
+    note: This vulnerability shows the importance of having multiple layers of security. Even if one function fails, others should be in place to prevent exploitation.
   least_privilege:
-    applies:
-    note:
+    applies: false
+    note: The vulnerability does not appear to be related to the principle of least privilege but rather to improper memory handling.
   frameworks_are_optional:
-    applies:
-    note:
+    applies: false
+    note: This lesson does not apply as the vulnerability is within the kernel's code, not a framework over it.
   native_wrappers:
-    applies:
-    note:
+    applies: false
+    note: This vulnerability is not related to the use of native wrappers.
   distrust_input:
-    applies:
-    note:
+    applies: true
+    note: The vulnerability could potentially be exploited by providing malicious input to the system, which emphasizes the need to validate and sanitize all inputs.
   security_by_obscurity:
-    applies:
-    note:
+    applies: false
+    note: Security by obscurity is not relevant to this vulnerability; it is a matter of proper error handling and memory management.
   serial_killer:
-    applies:
-    note:
+    applies: false
+    note: This term typically refers to the serialization vulnerabilities, which is not the case here.
   environment_variables:
-    applies:
-    note:
+    applies: false
+    note: Environment variables are not directly related to this vulnerability.
   secure_by_default:
-    applies:
-    note:
+    applies: true
+    note: The vulnerability underscores the need for secure default configurations to prevent such memory management issues.
   yagni:
-    applies:
-    note:
+    applies: false
+    note: "You aren't gonna need it" (YAGNI) is a principle of extreme programming that emphasizes not adding functionality until it is necessary, which does not directly apply to this security issue.
```

```suggestion
    note: |
      "You aren't gonna need it" (YAGNI) is a principle of extreme programming that 
      emphasizes not adding functionality until it is necessary, which does not directly 
      apply to this security issue.
```

TIL that starting a line with a quote messes up yaml!

---

Thank you so much for this information. I could not figure out why I kept getting the yaml error until I read your comment.